.github/workflows/task-defnition-cleanup.yaml

```diff
@@ -166,3 +167,127 @@ jobs:
+          set -e
+
+          REGION="$AWS_REGION"
+          DRY_RUN="$DRY_RUN"
+
+          echo "======================================================="
+          echo "Step 2: Delete all INACTIVE ECS Task Definitions (Paginated)."
+          echo "Region: $REGION"
+          echo "Dry run mode: $DRY_RUN"
+          echo "======================================================="
+
+          # Paginate manually over INACTIVE definitions
+          list_inactive_task_definitions() {
+            local next_token=""
+            local definitions=()
+
+            while : ; do
+              if [ -z "$next_token" ]; then
+                response=$(aws ecs list-task-definitions \
+                  --status INACTIVE \
+                  --region "$REGION" \
+                  --output json \
+                  --query '{taskDefinitionArns: taskDefinitionArns, nextToken: nextToken}')
+              else
+                response=$(aws ecs list-task-definitions \
+                  --status INACTIVE \
+                  --starting-token "$next_token" \
+                  --region "$REGION" \
+                  --output json \
+                  --query '{taskDefinitionArns: taskDefinitionArns, nextToken: nextToken}')
+              fi
+
+              current_batch=$(echo "$response" | jq -r '.taskDefinitionArns[]?')
+              if [ -n "$current_batch" ]; then
+                definitions+=( $current_batch )
+              fi
+
+              next_token=$(echo "$response" | jq -r '.nextToken // empty')
+              [ -z "$next_token" ] && break
+            done
+
+            echo "${definitions[@]}"
+          }
+
+          INACTIVE_TASKS_ARRAY=($(list_inactive_task_definitions))
+          TOTAL_INACTIVE=${#INACTIVE_TASKS_ARRAY[@]}
+
+          if [ "$TOTAL_INACTIVE" -eq 0 ]; then
+            echo "No INACTIVE task definitions found. Nothing to delete."
+            exit 0
+          fi
+
+          echo "Found $TOTAL_INACTIVE INACTIVE task definitions total."
+          echo "We'll delete them in chunks of up to 10."
+
+          # Function to delete up to 10 definitions (with backoff & jitter):
+          delete_chunk() {
+            local chunk=("$@")
+            echo "Deleting the following INACTIVE tasks:"
+            printf '%s\n' "${chunk[@]}"
+
+            for attempt in {1..5}; do
+              if aws ecs delete-task-definitions \
+                --task-definitions "${chunk[@]}" \
+                --region "$REGION"; then
+                echo "Successfully deleted chunk of up to 10 tasks."
+                break
+              else
+                echo "Attempt $attempt failed. Sleeping before retry..."
+                sleep $((attempt * 2))  # exponential backoff
+              fi
+
+              if [ "$attempt" -eq 5 ]; then
+                echo "ERROR: Failed to delete chunk after 5 attempts."
+                exit 1
+              fi
+            done
+
+            # Random jitter of 1–3s
```

Comment:
Weird - Why a random sleep time?

Reply:
To prevent AWS API throttling errors. This is making hundreds of API calls when it runs, s we want to make sure they are spaced apart.